packages/headless/src/controllers/result-list/headless-interactive-result.ts

```diff
@@ -79,6 +80,11 @@ export function buildInteractiveResult(
   };
 
   let wasOpened = false;
+
+  const addToRecentResultsList = () => {
+    engine.dispatch(pushRecentResult(options.result));
```

Comment:
This should probably be added into `logAnalyticsIfNeverOpened`

Reply:
Technically this isn't an analytics action though so I preferred to have it separate. And if the result is already in the list because it was opened before then it needs to be pushed anyway because it moves it to the top of the list